src/promptflow/promptflow/_core/token_provider.py

```diff
@@ -41,7 +41,21 @@ def _init_instance(self):
             # Initialize a credential instance
             from azure.identity import DefaultAzureCredential
 
-            self.credential = DefaultAzureCredential()
+            # Support sovereign cloud cases, like mooncake, fairfax.
+            from azure.ai.ml._azure_environments import (
+                _get_default_cloud_name, EndpointURLS, _get_cloud, AzureEnvironments
+            )
+
+            cloud_name = _get_default_cloud_name()
+            if cloud_name != AzureEnvironments.ENV_DEFAULT:
+                cloud = _get_cloud(cloud=cloud_name)
+                authority = cloud.get(EndpointURLS.ACTIVE_DIRECTORY_ENDPOINT)
+                self.credential = DefaultAzureCredential(
+                    authority=authority,
+                    exclude_shared_token_cache_credential=True
```

Comment:
why do we have to exclude_shared_token_cache_credential ?

Reply:
referred to this PR [[pfazure] support sovereign cloud](https://github.com/microsoft/promptflow/commit/f3dd31f0e4e4f863197496cf3ba5711883e84cea)